rust/lance/src/session.rs

```diff
@@ -94,9 +95,28 @@ impl Session {
         metadata_cache_size: usize,
         store_registry: Arc<ObjectStoreRegistry>,
     ) -> Self {
+        use lance_core::cache::SizedRecord;
+        use moka::future::Cache;
+
+        let index_cache_arc = Arc::new(
+            Cache::builder()
+                .max_capacity(index_cache_size as u64)
+                .weigher(|_, v: &SizedRecord| v.deep_size_of().try_into().unwrap_or(u32::MAX))
+                .support_invalidation_closures()
+                .build(),
+        );
```

Comment:
Why was this brought out of the LanceCache constructor?

Reply:
ah, good catch, this fell out of some rearranging I had previously done. Will address.